DEVELOPING.md

```diff
@@ -0,0 +1,36 @@
+# For Developers
+
+## Releasing
+
+Start from a clean checkout at `master`.
+
+Before running through the release it's good to run the build and the tests locally, and make sure CI is passing. You can
+also test-drive the commit in an existing Bazel workspace to sanity check functionality.
+
+#### Determining Semantic Version
+
+**rules_python** is currently using [Zero-based versioning](https://0ver.org/) and thus backwards-incompatible API
+changes still come under the minor-version digit. So releases with API changes and new features bump the minor, and
+those with only bug fixes and other minor changes bump the patch digit.   
+
+#### Steps 
+
+1. Update `version.bzl` with the new semantic version `X.Y.Z`.
+2. Run `bazel build //distro:rules_python-X.Y.Z` to build the distributed tarball.
+3. Calculate the Sha256 hash of the tarball. This hash will be used in the `http_archive` rules that download the new release.
+    1. Example command for OSX: `shasum --algorithm 256 bazel-bin/distro/rules_python-0.1.0.tar.gz`
```

Comment:
Slightly confusing that we generate a sha sum for the release and then embed it in the release commit. Shouldn't adding the new sha change the release sha?

Reply:
Yeah it is confusing I agree. It doesn't change the sha256 because the sha256 is added into the `examples/` directory which isn't included in the sha'd distributable. It a bit awkward to do it this way but it allows you to have one commit "release X.Y.Z" that contains the necessary version.bzl bump and also updates to the examples (and README possibly).

It could be more straightforward though to just do the examples/ update in a followup commit.

Reply:
nice `--algorithm` instead of `-a`